### ISSUE TITLE
Select widget and align horitonzally or vertically

### ISSUE THREAD

**Issue by simonaubertbd** [biolab/orange3#6477]
**What's your use case?**
On complex workflows, it's sometimes hard to follow visually the logic because widget boxes are not aligned.

**What's your proposed solution?**
The ability to align boxes, either horizontally, either vertically with the first selectiond widget box.

![image](https://github.com/biolab/orange3/assets/46959192/637ec695-d11a-4d4c-a2d3-7ae5f23e54e9)

**Are there any alternative solutions?**
Not that I know

**Comment by janezd** [comment]
This has been my wish for a long time as well, as well as of @BlazZupan, @markotoplak and probably others, too. I even tried to implement it, but it was out of my depth.

Even more than aligning widgets, we'd like to be able to arrange them so that lines between them would be horizontal. Also, we'd prefer "snapping", like in Illustrator or Keynote: when you drag a widget, it would snap to a position in which it is either aligned with another widget or the line is straight - of course, when it's close to that other widget and "almost" aligned.

When discussing it, some said that snapping would be enabled using some modifier key, I personally wouldn't mind if it's simply on all the time, or switchable through a setting (e.g. Cmd-U in Illustrator).

@ales-erjavec will look into it at his convenience.

**Comment by wvdvegte** [comment]
Yes, this would be a very nice enhancement. There is the 'Align widgets to grid' button, but the grid that it applies is too widely spaced, resulting in huge canvases for complex workflows
